agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/SpanDataMapper.java

```diff
@@ -410,16 +410,28 @@ private static String getTargetOrNull(Attributes attributes, int defaultPort) {
     if (peerService != null) {
       return peerService;
     }
-    String netPeerName = attributes.get(SemanticAttributes.NET_PEER_NAME);
-    if (netPeerName == null) {
-      return null;
+    String host = attributes.get(SemanticAttributes.NET_PEER_NAME);
+    if (host != null) {
+      Long port = attributes.get(SemanticAttributes.NET_PEER_PORT);
+      if (port != null && port != defaultPort) {
```

Comment:
Code blocks between lines 416/420 and 428/432 seem similar. What about an extract method? So the getTargetOrNull method could be shorter and faster to read?